tensorboard/webapp/widgets/line_chart_v2/line_chart_component_test.ts

```diff
@@ -648,4 +648,162 @@ describe('line_chart_v2/line_chart test', () => {
       expect(fixture.debugElement.query(By.css('.dark-mode'))).toBeTruthy();
     });
   });
+
+  describe('onContextLost renderer callback', () => {
+    function expectChartUpdateSpiesToHaveBeenCalledTimes(times: number) {
+      expect(setXScaleTypeSpy).toHaveBeenCalledTimes(times);
+      expect(setYScaleTypeSpy).toHaveBeenCalledTimes(times);
+      expect(updateMetadataSpy).toHaveBeenCalledTimes(times);
+      expect(updateDataSpy).toHaveBeenCalledTimes(times);
+      expect(updateViewBoxSpy).toHaveBeenCalledTimes(times);
+    }
+
+    it('does not recover the renderer by default', () => {
+      const fixture = createComponent({
+        seriesData: [buildSeries({id: 'foo'})],
+        seriesMetadataMap: {foo: buildMetadata({id: 'foo', visible: true})},
+        yScaleType: ScaleType.LINEAR,
+      });
+      fixture.detectChanges();
+      const chartBefore = fixture.componentInstance.chart.getLineChartForTest();
+
+      expectChartUpdateSpiesToHaveBeenCalledTimes(1); // Initial render.
+
+      fixture.componentInstance.chart.triggerContextLostForTest();
+      fixture.detectChanges();
+
+      expect(fixture.componentInstance.chart.getLineChartForTest()).toBe(
+        chartBefore
+      );
+      expect(disposeSpy).toHaveBeenCalledTimes(0);
+      expectChartUpdateSpiesToHaveBeenCalledTimes(1);
```

Comment:
Unclear whether these assertions are truly testing whether the line chart is attempting to recover. I guess assertion of `lineChartBefore === lineChartAfter` is the one? If so, put a semantic name on the assertion method so we can change the assertion easily.

Reply:
Good idea, done.